include/chainbase/chainbase.hpp

```diff
@@ -398,6 +400,16 @@ namespace chainbase {
             return _db_file.get_segment_manager()->get_free_memory();
          }
 
+         size_t get_reclaimable_memory() const {
+            size_t ret = 0;
+            for(const unique_ptr<abstract_index>& ai_ptr : _index_map) {
```

Comment:
Could we use `_index_list` instead of `_index_map`?

Reply:
I think so, yes, but we should also make the change elsewhere for consistency if doing that (I just copied what was being done elsewhere here). There likely isn't much of a benefit beyond skipping the `if(!ai_ptr)` check though (which _is_ nice to skip!), since the `_index_map` is still -- practically speaking -- small and just another `vector` anyways.

Reply:
Oh, OK, it was not clear to me that the `_index_map` was small as well, because of the comment at line 561:
`This is a full map (size 2^16) of all possible index designed for constant time lookup`.